### PR TITLE
refactor lib/private/Notification

### PR DESCRIPTION
**Pull request by shdehnavi**
## Summary
The required adjustments have been made to the classes in `/lib/private/Notification ` namespace.

The improvements:

- Using PHP8's constructor property promotion
- Adding return types
- Adding types to properties

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits

### CODE REVIEW

File: lib/private/Notification/Action.php [hunk 1]
```diff
@@ -28,22 +28,22 @@
 
 class Action implements IAction {
 	/** @var string */
```
Comment (artonge) [inline lib/private/Notification/Action.php:30]: You can remove those comment if they do not add more information

File: lib/private/Notification/Manager.php [hunk 2]
```diff
@@ -43,48 +43,35 @@
...
-								ICacheFactory $cacheFactory,
-								IRegistry $subscription,
-								LoggerInterface $logger,
-								Coordinator $coordinator) {
-		$this->validator = $validator;
-		$this->userManager = $userManager;
+	private bool $parsedRegistrationContext;
+
+	public function __construct(
+		protected IValidator      $validator,
```
Comment (artonge) [inline lib/private/Notification/Manager.php:67]: To much white spaces?

File: lib/private/Notification/Notification.php [hunk 3]
```diff
@@ -32,74 +32,72 @@
 use OCP\RichObjectStrings\IValidator;
 
 class Notification implements INotification {
-	/** @var IValidator */
-	protected $richValidator;
-
 	/** @var string */
```
Comment (artonge) [inline lib/private/Notification/Notification.php:35]: Same, some comments do not add more than the type declaration

File: lib/private/Notification/Action.php (hunk 1 above)
Comment (shdehnavi) [reply lib/private/Notification/Action.php]: It's addressed in the next commit

File: lib/private/Notification/Manager.php (hunk 2 above)
Comment (shdehnavi) [reply lib/private/Notification/Manager.php]: It's addressed in the next commit

File: lib/private/Notification/Notification.php (hunk 3 above)
Comment (shdehnavi) [reply lib/private/Notification/Notification.php]: It's addressed in the next commit

**Review by artonge — APPROVED** [review]

**Comment by shdehnavi** [comment]
@blizzz All conflicts have been resolved, Ready to merge

**Review by come-nc — APPROVED** [review]

**Comment by artonge** [comment]
Rebased to see if Cypress gets green. If not, can you try to check what's happening locally @shdehnavi?

**Comment by shdehnavi** [comment]
> Rebased to see if Cypress gets green. If not, can you try to check what's happening locally @shdehnavi?

I often encounter errors in Cypress jobs with most of my pull requests. I have rebased the code but nothing changed, still the same as earlier.

**Comment by artonge** [comment]
> I often encounter errors in Cypress jobs with most of my pull requests. I have rebased the code but nothing changed, still the same as earlier.

Docker image was outdated, I released a new version, Cypress tests are now green :)

**Comment by artonge** [comment]
Looks like drone is not triggered. And the performance step looks not related. Merging

**Pull request merged**